src/cowboy_websocket.erl

```diff
@@ -411,3 +412,10 @@ handler_terminate(#state{env=Env, handler=Handler},
 		Req, HandlerState, Reason) ->
 	cowboy_handler:terminate(Reason, Req, HandlerState, Handler),
 	{ok, Req, [{result, closed}|Env]}.
+	
+	
+%% @private
+get_real_socket({proxy_socket, _, Socket, _, _, _, _, _, _}) -> Socket;
```

Comment:
This is really not a great idea. We could be changing the structure any time.

The proper way to call it for us is with `Transport:match_port(Sock)` (as in https://github.com/heroku/ranch_proxy_protocol/blob/master/src/ranch_proxy_protocol.erl#L261-L262) -- this is non-standard and our own internal work-around, and we haven't done the due process of discussing with @essen as if that would be in any way a proper way to go about things.

So I figure that before pulling support in cowboy for this, we should check with @essen if he feels things should be done differently or if he at least agrees with our approach of allowing specific 'match ports', and also avoid internal matching like this.